### DISCUSSION TITLE
Can I store highlighted text areas and render the highlighted text programmatically?

### DISCUSSION THREAD

**Discussion by benlieb** [wojtekmaj/react-pdf#1476] [Q&A]
I'm new to the library and am confused by the use of the word `annotation`. We have a use case where we would like users to be able to select text, and write a comment about it. 

We would store the "address" of the highlight for rendering later, and the actual "comment" (what we call an annotation) in our database.

Can this library be used for something like this?

**Comment by wojtekmaj** [comment]
I'm not aware of any "unique ID" for a given text, let alone user-invoked selection. What you could do is create one yourself by combining document fingerprint + page index + text layer item index + selection start/end position within text layer item.

**Comment by benlieb** [comment]
This all makes sense. But I'm not sure what you mean by "document fingerprint". Is this the document id?

**Reply by wojtekmaj** [reply]
onDocumentLoad callback gets called with `document` you can get fingerprint from: `document._pdfInfo.fingerprint`.

In simple cases, you can also rely on other params that uniquely identify documents, e.g. their URLs.
